src/main/io/osd.c

```diff
@@ -684,6 +684,19 @@ static bool osdDrawSingleElement(uint8_t item)
                 break;
             }
 
+#ifdef USE_ESC_SENSOR
+			// Show warning if we lose motor output
+			if (enabledWarnings & OSD_WARNING_ESC_FAIL && ARMING_FLAG(ARMED)) {
+				for (int i = 0; i < getMotorCount(); i = i + 1) {
+					escSensorData_t *escDatai = getEscSensorData(i);
+					if (escDatai->rpm == 0) {
+				        tfp_sprintf(buff, "ESC %2d ERR", i + 1);
```

Comment:
Please use `osdFormatMessage()` to print your message here.

Also, you need to rebase this since the code you're based on is outdated.

Reply:
How do I use osdFormatMessage() with sprintf? sprintf to an intermediate buffer, and then osdFormatMessage()?

Reply:
Good point. That should work.